CHANGELOG.md

```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Improved layout.tree
 - Changed default theme color for repr.number to cyan
 - `__rich_measure__` signature changed to accept ConsoleOptions rather than max_width
+- `status` parameter to rich.status.Status changed to TextType
```

Comment:
Why do this? It works with any renderable.

```python
from rich.console import Console
from rich.panel import Panel
c=Console()
from time import sleep
with c.status(Panel("foo")):
    sleep(10)
```

Reply:
Wanted to see if this was viable change based on the docstrings and text type of the wrapped spinner object. That is, since the spinner text already expects a Text object and since the status object documentation seems to prefer text input, was hoping that this modification to status would be permitted.

The alternative solution would be to allow the spinner object to fully manage its own updating and text field would be to change the spinner object itself to take in a RenderableType. Sounds like that's what you'd prefer for this situation, so I've gone ahead and done that as the latest commit to this PR.

Reply:
Calling the Spinner arg "text" was probably an error on my part, but it would be a shame to prohibit other renderables since it works.